source/fundamentals/indexes.txt

```diff
@@ -0,0 +1,340 @@
+=======
+Indexes
+=======
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+.. common-content-begin
+
+Overview
+--------
+
+Indexes support the efficient execution of queries in MongoDB. Without indexes, MongoDB must scan *every* document in a
+collection (a **collection scan**) to find the documents that match each query. These collection scans are slow and can
+negatively affect the performance of your application. If an appropriate index exists for a query, MongoDB can use the
+index to limit the number of documents it must inspect.
+
+.. tip::
+
+   Indexes are also used by update operations when finding the document(s) to update, and by
+   :manual:`certain stages <core/aggregation-pipeline/#pipeline-operators-and-indexes>` in the aggregation framework.
+
+
+Query Coverage and Performance
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+When you execute a query against MongoDB, your command can include various elements:
+
+- query criteria that specify field(s) and value(s) you are looking for
+- options that affect the query's execution (e.g. read concern)
+- projection criteria to specify the fields MongoDB should return (optional)
+- sort criteria to specify the order documents will be returned from MongoDB (optional)
+
+When all the fields specified in the query, projection, and sort are indexed, MongoDB returns results directly from the
+index without scanning any documents in the collection or loading them into memory, also called a **covered query**.
```

Comment:
I would consider dropping "loading them into memory" as it is not clear what this means? If it is a refers to avoiding an in-memory sort then I think more context will be needed, as the index key order also has to match the sort specification order.

Reply:
Thanks, I've removed the terminology. In regards to your point about sorting, do you think the sort order note that follows this paragraph is sufficient in explaining this?